src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsintegrationapi/extensions/WebClientWrapper.kt

```diff
@@ -12,10 +12,10 @@ class WebClientWrapper(
     .baseUrl(baseUrl)
     .build()
 
-  inline fun <reified T> request(method: HttpMethod, uri: String, token: String, requestBody: Map<String, Any?>? = null): T {
+  inline fun <reified T> request(method: HttpMethod, uri: String, headers: Map<String, String>, requestBody: Map<String, Any?>? = null): T {
     val responseBodySpec = client.method(method)
       .uri(uri)
-      .header("Authorization", "Bearer $token")
+      .headers { header -> headers.forEach { h -> header.set(h.key, h.value) } }
```

Comment:
Maybe we name the parameter `requestHeader` and then we don't need to use just `h`?